pkg/server/api/enroll.go

```diff
@@ -124,7 +153,18 @@ func proxyAuthenticated(ap *config.AuthenticatingProxy, w http.ResponseWriter, r
 		return "", false
 	}
 
-	if !(clientAuthenticated(r) && clientHostnameMatches(ap.Hostname, r)) {
+	// Host name matching
+	hostnameMatches := (clientAuthenticated(r) && clientHostnameMatches(ap.Hostname, r))
+
+	// SPIFFE ID matching
+	spiffeIdMatches, err := clientSpiffeIdMatches(ap.SpiffeId, r)
+	if err != nil {
+		// Do not necessarily fail as other conditions might apply
+		logger.Error("malformed proxy spiffe id")
```

Comment:
maybe wrap this error, so we can still see what happened exactly? https://pkg.go.dev/errors has some examples